webui/src/lib/components/repository/changes.jsx

```diff
@@ -145,3 +151,103 @@ function useTreeItemType(entry, repo, leftDiffRefID, rightDiffRefID) {
     }, [leftResult, rightResult]);
     return treeItemType;
 }
+
+/**
+ * A container component for entries that represent diff between refs. this container is use by compare, commit changes,
+ * and uncommitted changes views.
+ *
+ * @param results to be displayed in the changes tree container
+ * @param showExperimentalDeltaDiffButton whether or not to display a delta-specific experimental feature button. TODO (Tals): remove when enabling the delta diff feature.
+ * @param delimiter objects delimiter ('' or '/')
+ * @param uriNavigator to navigate in the page using the changes container
+ * @param leftDiffRefID commitID / branch
+ * @param rightDiffRefID commitID / branch
+ * @param repo Repository
+ * @param reference commitID / branch
+ * @param internalRefresh to be called when the page refreshes manually
+ * @param prefix for which changes are displayed
+ * @param getMore to be called when requesting more diff results for a prefix
+ * @param loading of API response state to get changes
+ * @param nextPage of API response state to get changes
+ * @param setAfterUpdated state of pagination of the item's children
+ * @param onNavigate to be called when navigating to a prefix
+ * @param onRevert to be called when an object/prefix is requested to be reverted
+ */
+export const ChangesTreeContainer = ({results, showExperimentalDeltaDiffButton = false, delimiter, uriNavigator,
+                                         leftDiffRefID, rightDiffRefID, repo, reference, internalRefresh, prefix,
+                                         getMore, loading, nextPage, setAfterUpdated, onNavigate, onRevert}) => {
+    const [showComingSoonModal, setShowComingSoonModal] = useState(false);
+    const enableDeltaDiff = JSON.parse(localStorage.getItem(`enable_delta_diff`));
+    const sendDeltaDiffStats = async () => {
+        const deltaDiffStatEvents = [
+            {
+                "class": "experimental-feature",
+                "name": "delta-diff",
+                "count": 1,
+            }
+        ];
+        await statistics.postStatsEvents(deltaDiffStatEvents);
+    }
+
+    return <>
+        <div className="tree-container">
+            {(results.length === 0) ? <Alert variant="info">No changes</Alert> : (
+                <>
+                    {!enableDeltaDiff ?
+                        <>
+                        {showExperimentalDeltaDiffButton
+                            ? <>
```

Comment:
This should work, right?
```suggestion
                        { showExperimentalDeltaDiffButton && <>
```